File: be/src/main/java/sg/flow/entities/Account.java
package sg.flow.entities;

import java.time.LocalDateTime;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import sg.flow.entities.utils.AccountType;

@Data
@AllArgsConstructor
@Builder
@Table(name = "accounts")
public class Account {

    @Id
    private Long id;

    @NotNull
    private String accountNumber;

    @NotNull
    private Bank bank;

    @NotNull
    private User owner;

    @NotNull
    private Double balance;

    @NotNull
    private String accountName;

    @NotNull
    private AccountType accountType;

    @NotNull
    private Double interestRatePerAnnum;

    @NotNull
    private LocalDateTime lastUpdated;
}

File: be/src/main/kotlin/sg/flow/entities/Account.kt
package sg.flow.entities

import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Table
import sg.flow.entities.utils.AccountType

data class Account(
        @field:Id val id: Long,
        @field:NotNull val accountNumber: String,
        @field:NotNull val bank: Bank,
        @field:NotNull val owner: User,
        @field:NotNull val balance: Double,
        @field:NotNull val accountName: String,
        @field:NotNull val accountType: AccountType,
        @field:NotNull val interestRatePerAnnum: Double = 0.0,
        @field:NotNull val lastUpdated: LocalDateTime
)
